### PR TITLE
#164046244 Add rest API documentation

### PR DESCRIPTION
**Pull request by cob04**
#### What does this PR do?
Documents the RESTful API
#### Description of Task to be completed?
Add API documentation using `rest_framework.documentation`
#### How should this be manually tested?
* Follow instructions on setting up and running the app on `README.md` using the `develoment environment` in step 4.
* Checkout the branch `origin/ft-api-documentation-164046244`
* Visit the url `localhost:8000/swagger/` for swagger ui
* Visit the url `localhost:8000/redoc/` for redoc ui
#### Any background context you want to provide?
* Documentation is added using docstrings in the view functions or classes
#### What are the relevant pivotal tracker stories?
[#164046244](https://www.pivotaltracker.com/story/show/164046244)
#### Screenshots
* Swagger
screenshot_2019-03-01 legion author s haven

* ReDoc
screenshot_2019-03-01 legion author s haven 1

### CODE REVIEW

**Review by BrianSerem — APPROVED** [review]
All looks fine,

**Review by RachelleMaina — APPROVED** [review]

**Comment by MandelaK** [comment]
It looks okay, let me try and interact with the documentation and get back to you

File: README.md [hunk 1]
```diff
@@ -34,6 +34,10 @@ Authors Haven app can be run by:
     Run the app according to the environment you need:
 
    #### Development environment:
+    `python manage.py makemigrations --sttings=authors.settings.dev`
```
Comment (abulojoshua1) [inline README.md]: Check your spelling of settings on this line.

Comment (cob04) [reply README.md]: Thanks, I have fixed it.

**Review by Etomovich — APPROVED** [review]
LGTM

**Review by MandelaK — APPROVED** [review]

**Comment by ElMonstro** [comment]
The builds are failing because a typo in ignored end of file error code in the .flake8 file. Please change it to W391

**Comment by cob04** [comment]
@ElMonstro  OK let me have a look and fix it.

**Review by ElMonstro — APPROVED** [review]

**Pull request merged**